lib/audit_test.pm

```diff
@@ -47,78 +48,102 @@ sub run_testcase {
-    assert_script_run('./run.bash', %args);
+    assert_script_run("cd ${testdir}${testfile_tar}/audit-test/");
+    assert_script_run('make') if ($args{make});
+    assert_script_run("cd ${testcase}/");
+
+    # Export MODE
+    assert_script_run("export MODE=$audit_test::mode");
+
+    assert_script_run('./run.bash', timeout => $args{timeout});
 
     # Upload logs
     upload_logs("$current_file");
     upload_logs("$baseline_file");
 }
 
-# Parse all test cases in *.log (run.log, etc)
-# input: $file - file name
-# output: @testcase_list - test case list
-sub parse_testcase_log {
-    my ($file) = @_;
-
-    # Test case token, it is a number within []: e.g., [0], [1], [12]
-    my $s_tok         = '\[\d+\] ';
-    my @testcase_list = ();
-
-    my @lines = split(/\n/, path("$file")->slurp);
-    my $i     = 0;
-    # Parse the test cases of input file
-    for my $line (@lines) {
-        if ($line =~ /$s_tok/) {
-            push @testcase_list, "$line\n";
-            $i++;
+sub parse_lines {
+    my ($lines) = @_;
+    my @results;
+    foreach my $line (@$lines) {
+        if ($line =~ /(\[\d+\])\s+(.*)\s+(PASS|FAIL|ERROR)/) {
+            my $name = $2;
+            $name = trim $name;
+            push @results, {id => $1, name => $name, result => $3};
         }
     }
-    record_info("T: $i", "Total \"$i\" test cases found in file \"$file\":\n @testcase_list");
-    return @testcase_list;
+    return @results;
 }
 
 # Compare baseline testing result and current testing result
 # input: $testcase - test case name (the test module name in openQA code,
 # if test module is 'audit_tools.pm' then $testcase = audit_tools, etc)
 sub compare_run_log {
     my ($testcase) = @_;
-    my $c_file     = "ulogs/${testcase}-${current_file}";
-    my $b_file     = "ulogs/${testcase}-${baseline_file}";
-
-    # Define Test case (name and result) list
-    my @testcase_list_current  = ();
-    my @testcase_list_baseline = ();
-    my $result                 = 'ok';
-
-    # Parse the test cases in baseline file
-    record_info('Baseline', 'Parse baseline test results');
-    @testcase_list_baseline = parse_testcase_log($b_file);
-    record_info('Current', 'Parse current test results');
-    @testcase_list_current = parse_testcase_log($c_file);
-    my $str_baseline = join('', @testcase_list_baseline);
-    my $str_current  = join('', @testcase_list_current);
-    record_info('Compare', "Compare the testing results of current file \"$c_file\" with baseline file \"$b_file\".");
-    if ($str_baseline eq $str_current) {
-        $result = 'ok';
+
+    # Read the current test result from rollup.log
+    my $output          = script_output('cat ./rollup.log');
+    my @lines           = split(/\n/, $output);
+    my @current_results = parse_lines(\@lines);
+
+    my %baseline_results;
+    my $baseline_file = "ulogs/$testcase-baseline_run.log";
+    if (!-e $baseline_file) {
+        diag "The file $baseline_file does not exist";
     }
```

Comment:
```suggestion
    diag "The file $baseline_file does not exist" unless -e $baseline_file;
```

Reply:
we also have an `else` here. If I change that to `diag "The file $baseline_file does not exist" unless -e $baseline_file;`,  we still need to change `else`. So I prefer to keep that.